tutorials/ROS_Integration/ros_ign_demo.md

```diff
@@ -0,0 +1,84 @@
+# ROS Integration
+
+In this tutorial we will learn how to Integrate ROS with Ignition. We will establish
+communication between them. This can help in many aspects; we can receive data or commands
+from ROS and apply it to Ignition and vice versa.
+
+## ros_ign_bridge
+
+`ros_ign_bridge` provides a network bridge which enables the exchange of messages between ROS 2 and Ignition Transport. Its support is limited to only certain message types. Please, check this [README](https://github.com/ignitionrobotics/ros_ign/blob/ros2/ros_ign_bridge/README.md) to verify if your message type is supported by the bridge.
+
+## Requirements
+
+For this tutorial to work correctly make sure you have the following installed:
+
+* [ROS 2 Foxy](https://index.ros.org/doc/ros2/Installation/Foxy/)
+* [Ignition Citadel](https://ignitionrobotics.org/docs/citadel)
+* [ros_ign_bridge](https://github.com/ignitionrobotics/ros_ign/blob/ros2/ros_ign_bridge/README.md#building-the-bridge-from-source)
+
+## Bidirectional communication
+
+We can initialize a bidirectional bridge so we can have ROS as the publisher and Ignition as the subscriber or vice versa.
+
+For example:
+
+```
+ros2 run ros_ign_bridge parameter_bridge /TOPIC@ROS_MSG@IGN_MSG
+```
+
+The `ros2 run ros_ign_bridge parameter_bridge` command simply runs the `parameter_bridge` code from the `ros_ign_bridge` package. Then, we specify our topic `/TOPIC` over which the messages will be sent. The first `@` symbol delimits the topic name from the message types. Following the first `@` symbol is the ROS message type.
+
+The ROS message type is followed by an `@`, `[`, or `]` symbol where:
+
+* `@`  is a bidirectional bridge.
+* `[`  is a bridge from Ignition to ROS.
+* `]`  is a bridge from ROS to Ignition.
+
+Have a look at these [examples](https://github.com/ignitionrobotics/ros_ign/blob/ros2/ros_ign_bridge/README.md#example-1a-ignition-transport-talker-and-ros-2-listener)
+explaining how to make communication connections from ROS to Ignition and vice versa.
+
+## Publish key strokes to ROS
+
+Let's send messages to ROS using the `Key Publisher` an Ignition Gazebo plugin.
+
+**Note:** Make sure to have ROS, Ignition and the bridge workspace sourced.
+
+* For ROS:  `. /opt/ros/foxy/setup.bash`
+* For Ignition (for source installation only):   `. ~/workspace/install/setup.bash`
+* For bridge workspace:  `. ~/bridge_ws/install/setup.bash`
```

Comment:
Since we don't explain how to compile the bridge, let's not confuse the user mentioning a workspace that they may not have. If they compiled the bridge from source, they will have instructions to source it there. Just a reminder here is enough.
```suggestion
```